pkg/chartutil/validate_name.go

```diff
@@ -53,6 +53,9 @@ var (
 
 const (
 	// maxNameLen is the maximum length Helm allows for a release name
+	// As of Kubernetes 1.4, the max limit on a name is 63 chars. We reserve 10 for
+	// charts to add data. Effectively, that gives us 53 chars.
+	// See https://github.com/helm/helm/issues/1528
```

Comment:
I would replace this comment with the following as its more accurate now:

```go
\\ According to the Kubernetes docs (https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#rfc-1035-label-names)
\\ some resource names have a max length of 63 characters while others have a max
\\ length of 253 characters. As we cannot be sure the resources used in a chart, we
\\ therefore need to limit it to 63 chars and reserve 10 chars for additional part to name
\\ of the resource. The reason is that chart maintainers can use release name as part of
\\ the resource name (and some additional chars).

```

Reply:
Okay, updated comments.